cou/utils/juju_utils.py

```diff
@@ -198,6 +198,15 @@ def is_from_charm_store(self) -> bool:
         """
         return self.origin == "cs"
 
+    def get_units_to_run_action(self) -> list[Unit | SubordinateUnit]:
+        """Get the unit or subordinate units in the application to run action.
+
+        :return: The unit or subordinate units
+        :rtype: List of Unit or SubordinateUnit
+        """
+        units = self.subordinate_units if self.is_subordinate else self.units.values()
+        return units  # type: ignore[return-value]
```

Comment:
Why do we need to ignore the return type?

Reply:
it kept complaining about return type should be `Collection[object]`, even though it seems correct: `list[Unit | SubordinateUnit]`

Reply:
Ah - that's because `units.values()` is a dict values class, not a list.  I guess `Collection` is a superset of list and dict values.

Reply:
I try doing `self.subordinate_units if self.is_subordinate else list(self.units.values())`, and same

Reply:
Strange, I don't know :thinking: